Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -164,6 +183,30 @@ + (void)clearConfiguration
     globalConfiguration() = nil;
 }
 
++ (void)maybeConsumeBundlePaths:(NSDictionary<NSAttributedStringDocumentReadingOptionKey, id> *)options
+{
+    auto* readAccessFileURLs = dynamic_objc_cast<NSArray<NSURL *>>(options[_WKReadAccessFileURLsOption]);
+    if (!readAccessFileURLs)
+        return;
+
+    if (!readAccessFileURLs.count)
+        return;
+
+    if (readAccessFileURLs.count > maximumReadOnlyAccessPaths)
+        [NSException raise:NSInvalidArgumentException format:@"_WKReadAccessFileURLsOption may have at most two additional directories."];
+
+    for (id fileURL in readAccessFileURLs) {
+        if (!dynamic_objc_cast<NSURL>(fileURL))
+            return;
```

Comment:
We throw an exception when the count is too high. What other types of errors should be throwing exceptions rather than silently doing nothing? Also, maybe we should always clear the old paths even if we do throw an exception?

Reply:
It seems reasonable to throw if we are given incorrect types (e.g., NSString* paths instead of NSURL*).

We can also throw if the NSURLs are for non-file URLS (which are the only types this API is meant to support).

Clearing the cached state if incorrect arguments are passed seems like a good idea. I'll add that and a test case.